src/Suave/Sockets/TcpTransport.fs

```diff
@@ -3,13 +3,29 @@
 open System
 open System.Net.Sockets
 
-type TcpTransport =
-  { socket    : Socket
-    readArgs  : SocketAsyncEventArgs
-    writeArgs : SocketAsyncEventArgs
-  }
+type TcpTransport(acceptArgs: SocketAsyncEventArgs, a : ConcurrentPool<SocketAsyncEventArgs>,b : ConcurrentPool<SocketAsyncEventArgs>,c : ConcurrentPool<SocketAsyncEventArgs>) =
+  let socket = acceptArgs.AcceptSocket
+  let readArgs = b.Pop()
+  let writeArgs = c.Pop()
+  let shutdownSocket _ =
+    try
+      if socket <> null then
+        try
+          socket.Shutdown(SocketShutdown.Both)
+        with _ -> ()
+        socket.Close ()
+        socket.Dispose ()
+    with _ -> ()
```

Comment:
To consider in the future: have a counter checking exceptions? Or we could use the System.Diagnostics global counter during a stress tests; because if this happens a lot, it can be a drag on Suave's performance.